server/src/main/java/com/ca/lsp/cobol/service/MyTextDocumentService.java

```diff
@@ -276,6 +280,16 @@ private void notifyAnalysisFinished(String uri, Map<String, List<Location>> copy
             .build());
   }
 
+  @Override
+  public CompletableFuture<List<Either<SymbolInformation, DocumentSymbol>>> documentSymbol(DocumentSymbolParams params) {
+    String uri = params.getTextDocument().getUri();
+    return outlineMap.get(uri)
+        .thenApply(documentSymbols -> documentSymbols.stream()
```

Comment:
Will it work if we receive this request two times for one analysis? I.e. when we receive `didOpen` and `outline`, then `didChange` and `outline` again, but at the time when the `didChange` has not put the new feature to the map?

Reply:
I think since the didChange is doing is stuff on the client side there is the "old" version of the outline view that should be replaced on the client (I am testing that behavior)

Reply:
Here is the simple test for future:
```
  @Test
  void test() throws ExecutionException, InterruptedException {
    CompletableFuture<Integer> cf = new CompletableFuture<>();
    cf.complete(42);
    System.out.println(process(cf));
    System.out.println(process(cf));
  }

  int process(CompletableFuture<Integer> cf) throws ExecutionException, InterruptedException {
    return cf.thenApply(i -> i*2).get();
  }
```
It works well. I think that it will be safe to resolve the Future twice.